openedx/core/lib/block_structure/transformers.py

```diff
@@ -105,3 +110,40 @@ def is_collected_outdated(cls, block_structure):
             )
 
         return bool(outdated_transformers)
+
+    def _transform_optimized(self, transformers, block_structure):
+        """
+        Transforms the given block_structure using the transform_block_filter
+        method from the given transformers.
+        """
+        if not transformers:
+            return
+
+        removal_filters = []
+        for removal in [t.transform_block_filter(self.usage_info, block_structure) for t in transformers]:
+            if isinstance(removal, list):
```

Comment:
You could even avoid creating multiple lists, and make `transform_block_filter()` a generator that yields results one by one.